### PR TITLE
[CPM-162] Select Courses from UoA API

### PR DESCRIPTION
**Pull request by FocalChord**
## Changes

- Checkbox List to select courses to pick
- Searchable List
- API Changed 
- State Management for overriding 
- CSS Changes

## Checklist

- [x] Have tests been written?
- [x] Is the UI consistent?

Addresses [CPM-162]

[CPM-162]: https://momentum-agile.atlassian.net/browse/CPM-162

### CODE REVIEW

**Review by daltonlim — APPROVED** [review]
lgtm. Just wondering if the shown courses should be alphabetically ordered

File: api/src/routes/uni-api-route.js [hunk 1]
```diff
@@ -1,157 +1,238 @@
...
+const { parseRequirements } = require("../parser");
+
+// Client posts the courses they want and then only those are saved in the database
+/**
+ * @swagger
+ * path:
+ *  /uni/programme:
+ *    post:
+ *      tags: [Uni]
+ *      summary: Get a plan by plan code
```
Comment (Richardng12) [inline api/src/routes/uni-api-route.js:15]: Fix swagger documentation

File: api/src/routes/uni-api-route.js [hunk 2]
```diff
@@ -1,157 +1,238 @@
...
+
+// To get the courses for a certain programme
+
+/**
+ * @swagger
+ * path:
+ *  /uni/programme:
+ *    get:
+ *      tags: [Uni]
+ *      summary: Get a plan by plan code
```
Comment (Richardng12) [inline api/src/routes/uni-api-route.js:113]: Here asw

File: api/src/routes/uni-api-route.js [hunk 3]
```diff
@@ -1,157 +1,238 @@
...
+});
+
+// GET courses for a particular course number
+/**
+ * @swagger
+ * path:
+ *  /uni/course:
+ *    get:
+ *      tags: [Uni]
+ *      summary: Get a plan by plan code
```
Comment (Richardng12) [inline api/src/routes/uni-api-route.js:174]: Same thing

**Review by Richardng12 — APPROVED** [review]
Lgtm

**Pull request merged**